src/Infrastructure/Identity/TokenService.cs

```diff
@@ -23,23 +25,46 @@ internal class TokenService : ITokenService
     private readonly MailSettings _mailSettings;
     private readonly JwtSettings _jwtSettings;
     private readonly FSHTenantInfo? _currentTenant;
+    private readonly ICaptchaService _capchaService;
+    private readonly IConfiguration _config;
 
     public TokenService(
         UserManager<ApplicationUser> userManager,
         IOptions<JwtSettings> jwtSettings,
         IStringLocalizer<TokenService> localizer,
         IOptions<MailSettings> mailSettings,
-        FSHTenantInfo? currentTenant)
+        FSHTenantInfo? currentTenant,
+        ICaptchaService captchaService,
+        IConfiguration configuration)
     {
         _userManager = userManager;
         _localizer = localizer;
         _mailSettings = mailSettings.Value;
         _jwtSettings = jwtSettings.Value;
         _currentTenant = currentTenant;
+        _capchaService = captchaService;
+        _config = configuration;
     }
 
     public async Task<TokenResponse> GetTokenAsync(TokenRequest request, string ipAddress, CancellationToken cancellationToken)
```

Comment:
ipAddress ??? where do you get this field?

Reply:
it comes from tokens controller with request in this repo.